### PR TITLE
Introduce Signer interface for key management (breaking changes, moving to SDK 2.0)

### PR DESCRIPTION
**Pull request by netoneko**
Before:

```javascript
const sender = Orbs.createAccount();
const receiver = Orbs.createAccount();
const orbsClient = new Orbs.Client(endpoint, virtualChainId, "TEST_NET");

const [tx, txId] = orbsClient.createTransaction(
  sender.publicKey, sender.privateKey, // removed in 2.0
  "BenchmarkToken", "transfer", 
  [Orbs.argUint64(10), Orbs.argAddress(receiver.address)]);

const query = orbsClient.createQuery(sender.publicKey, 
"BenchmarkToken", "getBalance", 
[Orbs.argAddress(receiver.address)]);
```

After:
```javascript
const sender = Orbs.createAccount();
const receiver = Orbs.createAccount();
const senderClient = new Orbs.Client(endpoint, virtualChainId, "TEST_NET", 
  new Orbs.DefaultSigner(sender)); // added in 2.0

const [tx, txId] = await senderClient.createTransaction("BenchmarkToken", "transfer", 
  [Orbs.argUint64(10), Orbs.argAddress(receiver.address)]); // became async in 2.0

const query = await senderClient.createQuery("BenchmarkToken", "getBalance", 
  [Orbs.argAddress(receiver.address)]); // became async in 2.0
```

Notice that references to `publicKey` and `privateKey` were removed and instead you have to pass `new DefaultSigner({publicKey, privateKey})` which encapsulates signing.

The purpose of this change is to be able to inject the signer from the outside without exposing the keys, a step forward to implementing a wallet like Metamask.

### CODE REVIEW

**Comment by electricmonk** [comment]
My only qualm is around the `DefaultSigner` name - can we name it `LocalSigner` instead? or a different name that reflects what distinguishes it from other future signers?

File: e2e/browser/browser.test.js [hunk 1]
```diff
@@ -57,19 +57,19 @@ describe("E2E browser", () => {
...
-  });
-
   it("should create the sender account", async () => {
     await clickOnElement("#create-sender-account");
     const senderAccountId = await getElementText("#sender-account-id");
     const accountIdLength = senderAccountId.split(",").length;
     expect(accountIdLength).toEqual(32);
   });
 
+  it("should create Orbs.Client instance", async () => {
```
Comment (gilamran) [inline e2e/browser/browser.test.js]: These lines were just moved, right?

**Comment by gilamran** [comment]
I agree with Shay, `DefaultSigner` is too general. I also like the `LocalSigner`.

**Comment by gilamran** [comment]
The file `Signature` can be better described by calling it `Signer` instead.

**Comment by gilamran** [comment]
I don't understand the why you erased the contracts clone, and used a local versions of it (input.json and output.json). Without the clone you might miss changes

**Comment by netoneko** [comment]
I have removed the contracts clone because it's not really compatible with the old format anymore (because keys moved to a different place in the config as they are not parameters of the calls anymore).

We won't miss the changes because we run this e2e against gamma, so there is no difference if the input jsons come from outside or are stored here and can be refactored.

File: e2e/browser/browser.test.js (hunk 1 above)
Comment (netoneko) [reply e2e/browser/browser.test.js]: Yeah. I just moved to async/await syntax.

**Comment by netoneko** [comment]
I believe I have resolved all the issues.

**Comment by gilamran** [comment]
Great job!!

**Pull request merged**